### PR TITLE
Some fixes, added simple frontend framework

### LINKED ISSUE #20: [FEATURE] Little frontend framework into GoBlitz
**Is your feature request related to a problem? Please describe.**
Right now the frontend development in GoBlitz is very tricky and hard. Make it easier, by using helpers, views and middleware.

**Describe the solution you'd like**
Setup the frontend development framework based on:

- Isolated assets + common assets
- Components
- Go HTML Templates

**Note** Under assets are meant JS, CSS, IMG, GIF etc.

**Describe alternatives you've considered**
Make GoBlitz as an backend web framework only

**Additional context**
-

### PR DESCRIPTION
**Pull request by KostLinux**
# Description

Added simple frontend development framework "GoBlitz Native" and made an documentation for him - 

Closes #20 

## Dependencies
None

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Tested locally multiple times
- [x] Revalidated code via `make validate`

**Test Configuration**:
* Device: MacOS
* Tools (optional):
* SDK (optional):

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas (optional)
- [x] I have made corresponding changes to the documentation (optional)
- [x] My changes generate no new warnings
- [x] I've performed/added tests/lints that prove my fix is effective or that my feature works (optional)

Before creating pull request, look that all points under checklist are done!

### CODE REVIEW

File: helpers/file.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+func CheckFileExists(path string) bool {
+	_, err := os.Stat(path)
+	return !os.IsNotExist(err)
+}
+
+func CheckFileNotExists(path string) bool {
+	_, err := os.Stat(path)
+	return os.IsNotExist(err)
+}
```
Comment (Matrix278) [inline helpers/file.go:13]: Why do you need 2 identical functions just by reversing? If you use CheckFileExists which returns boolean, you could use NOT CheckFileExists which still returns boolean but reversed.

Maybe rename as well, like CheckFileExists → CheckIfFileExists

File: middleware/router.go [hunk 2]
```diff
@@ -56,11 +61,6 @@ func NewRouter(router *gin.Engine) *gin.Engine {
 		apiGroup.GET("/users", api.GetUsers)
 	}
 
-	// HTML Templates (e.g Status page)
-	router.LoadHTMLGlob("./views/**/*")
-	router.GET("/status", httpTemplates.StatusPageResponse(), StatusPageMiddleware())
-	router.GET("/docs", DocumentationPageMiddleware())
-
```
Comment (Matrix278) [inline middleware/router.go:63]: Don't you need this page anymore?

Comment (Matrix278) [reply middleware/router.go:63]: Ah, I think I can see, you deleted it and did on some domain

Comment (KostLinux) [reply middleware/router.go:63]: They are moved upper, the docs is also changed. Header component in views:
```
                        <li><a href="https://docs.goblitz.org/" class="text-white ml-8">Documentation</a></li>
```
And the middleware/router.go. You should see in branch that:
```
	// Frontend & Static Files Handling
	router.LoadHTMLGlob("./views/**/*.html")

	router.Use(static.Serve("/assets", static.LocalFile("./public/assets", true)))
	if helper.CheckFileNotExists("./public/index.html") {
		router.GET("/", WelcomePageMiddleware())
		router.GET("/favicon.ico", func(c *gin.Context) {
			c.String(http.StatusNoContent, "")
		})
	} else {
		router.Use(static.Serve("/", static.LocalFile("./public/", true)))
	}

	helper.ServePageAssets(router)

	// HTML Template generated pages
	router.GET("/status", httpTemplates.StatusPageResponse(), StatusPageMiddleware())
```
The router.LoadHTMLGlob(".views/**/*") is a bug, which makes trouble

File: helpers/file.go (hunk 1 above)
Comment (KostLinux) [reply helpers/file.go]: Agree, fixed in [db9a3bf](https://github.com/LookinLabs/GoBlitz/pull/21/commits/db9a3bf1cf360aaba5483cc75761ef4795fc5c48)

**Review by Matrix278 — APPROVED** [review]

**Pull request merged**